utils/client_helper.go

```diff
@@ -334,6 +336,67 @@ func HasFIPS(ctx context.Context, client runtimeclient.Client) (bool, error) {
 	return machineConfig.Spec.FIPS, nil
 }
 
+func GetAdditionalTrustBundleFromConfigmap(ctx context.Context, client client.Client, configmapName string) (string, error) {
+	userCaBundleConfigmap := corev1.ConfigMap{}
+	if err := client.Get(ctx, types.NamespacedName{Name: common.ClusterAdditionalTrustBundleName,
+		Namespace: common.OpenshiftConfigNamespace}, &userCaBundleConfigmap); err != nil {
+		if errors.IsNotFound(err) {
+			return "", nil
+		}
+
+		return "", fmt.Errorf("failed to get %s/%s configmap: %w", common.OpenshiftConfigNamespace, common.ClusterAdditionalTrustBundleName, err)
+	}
+
+	if userCaBundleConfigmap.Data == nil {
+		return "", nil
+	}
+
+	if userCaBundleConfigmap.Data[common.CaBundleDataKey] == "" {
+		return "", nil
+	}
+
+	return userCaBundleConfigmap.Data[common.CaBundleDataKey], nil
+}
+
+func GetClusterAdditionalTrustBundleState(ctx context.Context, client client.Client) (bool, string, error) {
+	clusterAdditionalTrustBundle, err := GetAdditionalTrustBundleFromConfigmap(ctx, client, common.ClusterAdditionalTrustBundleName)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to get additional trust bundle from configmap: %w", err)
+	}
+
+	hasUserCaBundle := clusterAdditionalTrustBundle != ""
+
+	proxy := ocp_config_v1.Proxy{}
+	if err := client.Get(ctx, types.NamespacedName{Name: common.OpenshiftProxyCRName}, &proxy); err != nil {
+		return false, "", fmt.Errorf("failed to get proxy: %w", err)
+	}
+
+	proxyCaBundle := ""
+	switch proxy.Spec.TrustedCA.Name {
+	case common.ClusterAdditionalTrustBundleName:
+		proxyCaBundle = clusterAdditionalTrustBundle
+	case "":
+		// No proxy trustedCA configmap is set, do nothing
+	default:
+		proxyCaBundle, err = GetAdditionalTrustBundleFromConfigmap(ctx, client, proxy.Spec.TrustedCA.Name)
```

Comment:
if `configmapName` removed from above suggestion, we may need to remove this one as well
```suggestion
		proxyCaBundle, err = GetAdditionalTrustBundleFromConfigmap(ctx, client)
```